include/Gaffer/PlugAlgo.h

```diff
@@ -71,6 +71,12 @@ bool canPromote( const Plug *plug, const Plug *parent = NULL );
 /// related to plug layouts are copied to the external
 /// plug - this can be controlled with the `excludeMetadata`
 /// argument.
+///
+/// > Note : When promoting to a Box, this function examines
+/// > the value of the "nodule:type" metadata, and automatically
+/// > creates an intermediate BoxIO node if the plug has a
+/// > nodule. This makes the exact effects of the function dependent
+/// > on whether or not the UI metadata has been loaded.
```

Comment:
A Lighting BRP (fancy Box) from a recent show had 65 exposed non-nodule plugs (almost exclusively bools, strings, and floats for what its worth). For this particular example, they were all made from the UIEditor and then connected down inside, but I don't think that most artists work that way. It had 4 nodule plugs (all of which come from the node definition and not through promotion, though we'll likely update that to use the BoxIOs as well). Grepping the file, there were 25094 plug constructors in the gfr in total, 7275 of which have serialized nodule:type metadata. Of those, 378 are `GafferUI::StandardNodule`, 6891 are `''`, and 6 are `None`.

Talking with Lookdev, they thought 10-20 non-nodule plugs was normal for a LookdevBox and rarely promote nodule plugs (just getting the scene in/out pair plus maybe a filter). For their internal lookdev tools (also Box/References), they do expose more nodules (shader inputs), but rarely more than 5-10.